semantic_router/index/pinecone.py

```diff
@@ -584,5 +596,101 @@ async def _async_describe_index(self, name: str):
         async with self.async_client.get(f"{self.base_url}/indexes/{name}") as response:
             return await response.json(content_type=None)
 
+    async def _async_get_all(
+        self, prefix: str | None = None, include_metadata: bool = False
```

Comment:
can't use these type annotations as we need to remain backwards compatible with python 3.9, should instead use `Optional[str] = None` in this case